application/src/main/java/org/opentripplanner/graph_builder/module/osm/WalkableAreaBuilder.java

```diff
@@ -185,49 +168,41 @@ public void buildWithVisibility(AreaGroup group) {
     // List of edges belonging to the walkable area
     Set<Edge> edges = new HashSet<>();
 
-    // Edges which are part of the rings. We want to keep there for linking even tough they
-    // might not be part of the visibility edges.
+    // Edges which are part of the rings
     Set<Edge> ringEdges = new HashSet<>();
 
     // OSM ways that this area group consists of
     Set<Long> osmWayIds = group.areas
       .stream()
       .map(area -> area.parent)
-      .flatMap(osmWithTags ->
-        osmWithTags instanceof OsmRelation
-          ? ((OsmRelation) osmWithTags).getMembers().stream().map(OsmRelationMember::getRef)
-          : Stream.of(osmWithTags.getId())
+      .flatMap(osmEntity ->
+        osmEntity instanceof OsmRelation
+          ? ((OsmRelation) osmEntity).getMembers().stream().map(OsmRelationMember::getRef)
```

Comment:
I'm not sure if it's possible but can you try to use a pattern variable here to avoid the cast?

Reply:
seems to work!